### ISSUE TITLE
New plugin suggestion

### ISSUE THREAD

**Issue by gnudles** [LMMS/lmms#4641]
You know, I think it can be nice to have a JavaScript version of the X-pressive plug-in. It can use the built-in JS support in Qt via QtScript. The performance issue should be considered, but it seems a very nice idea.
Recently, I did a demo with Qt outside of LMMS to play a note with QtScript (JS), and it responded well.
What do you think?

**Comment by musikBear** [comment]
I am against any openings that could be used for malicious purposes. I think the X-pressive maths is sufficient for the purpose.

**Comment by SecondFlight** [comment]
@musikBear, Javascript does not open a vulnerability by itself, since there is no networking involved.

**Comment by gnudles** [comment]
I agree that we must to block some "features" in some manner. Like accessing the "System" object... it's not needed for a pure mathematical calculation.

**Comment by gnudles** [comment]
I agree that we must to block some "features" in some manner. Like accessing the "System" object... it's not needed for a pure mathematical calculation.

**Comment by musikBear** [comment]
@SecondFlight still possible to make all kind of idiotics that can make inexperienced pc-owners believe all is ruined. It is not 'just' real severe issues i think of, it is also all types of 'pranks', harmless as they may be, they will lead to a lot of noise in forum, and unnecessary  waste of time.

**Comment by SecondFlight** [comment]
I think I see what you're saying, certainly adding an arbitrary scripting language could introduce plenty of issues of that nature.
